fs/fs.go

```diff
@@ -0,0 +1,644 @@
+			if forSubDirs {
+				subdirs, _ := ioutil.ReadDir(p)
+				for _, sd := range subdirs {
+					dirs = append(dirs, path.Join(p, sd.Name()))
+				}
+			} else {
+				dirs = append(dirs, paths...)
+			}
+		}
+
+		if len(dirs) > 0 {
+			for {
+				for _, d := range dirs {
+					size, err := diskUsage(d)
+					if err != nil {
+						fmt.Fprintf(os.Stderr, "WORKER %s, ERROR {%s} for %s\n", id, err, d)
+						break
+					}
+					c.Mut.Lock()
+					c.DiskUsage[d] = size
+					c.Mut.Unlock()
+				}
+				time.Sleep(30 * time.Second)
+			}
+		} else {
+			fmt.Fprintf(os.Stderr, "WORKER %s, ERROR no storage points to collect", id)
+		}
+	}(forSubDirs, id, paths...)
+}
+
+func diskUsage(dir string) (uint64, error) {
+	out, _ := exec.Command("du", "-s", dir).Output()
+	val := strings.Fields(string(out))[0]
+	size, err := strconv.ParseUint(val, 10, 64)
+	if err != nil {
+		return 0, err
+	}
+
+	return size, nil
+}
+
+type partition struct {
+	mountpoint string
+	major      uint
+	minor      uint
+	fsType     string
+	blockSize  uint
+}
+
+type RealFsInfo struct {
+	// Map from block device path to partition information.
+	partitions map[string]partition
+	// Map from label to block device path.
+	// Labels are intent-specific tags that are auto-detected.
+	labels map[string]string
+
+	dmsetup dmsetupClient
+}
+
+type Context struct {
+	// docker root directory.
+	Docker  DockerContext
+	RktPath string
+}
+
+type DockerContext struct {
+	Root         string
+	Driver       string
+	DriverStatus map[string]string
+}
+
+func GetFsStats(container *docker.Container) (map[string]wrapper.FilesystemInterface, error) {
+	var (
+		baseUsage           uint64
+		logUsage            uint64
+		rootFsStorageDir    = storageDir
+		logsFilesStorageDir string
+	)
+
+	fsStats := map[string]wrapper.FilesystemInterface{}
+
+	if container.ID != "" {
+		getUserLayerId := func(storageDir, storageDriver, containerId string) (string, error) {
+			dockerVersion := config.DockerVersion
+			if dockerVersion[0] <= userLayerFirstVersionMaj && dockerVersion[1] < userLayerFirstVersionMin {
+				return containerId, nil
+			}
+			switch storageDriver {
+			case aufsStorageDriver:
+				fallthrough
+			case overlayStorageDriver:
+				idFilePath := filepath.Join(storageDir, "image", storageDriver, "layerdb", "mounts", containerId, userLayerIdFile)
+				idBytes, err := ioutil.ReadFile(idFilePath)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "Failed to read id of user-layer for container  %v from under path  %v\n", containerId, idFilePath)
+					return "", err
+				}
+				return string(idBytes), nil
+			default:
+				return "", fmt.Errorf("Unsupported storage driver; dont know how to determine id of user layer for container %v \n", containerId)
+			}
+		}
+		userLayerId, err := getUserLayerId(storageDir, container.Driver, container.ID)
+		if err != nil {
+			userLayerId = container.ID
+		}
+
+		switch container.Driver {
+		case aufsStorageDriver:
+			// `/var/lib/docker/aufs/diff/<docker_id>`
```

Comment:
I suggest adding additional information in this comment as it is in [line](https://github.com/intelsdi-x/snap-plugin-collector-docker/pull/32/files#diff-f6c6bb8a6ec40b9f63d107bc7070f708R190) line. It will be easier to understand why this comment is needed here.